### DISCUSSION TITLE
Sono 4 pro last switch doesn't work

### DISCUSSION THREAD

**Discussion by crazyclimber999** [RavenSystem/esp-homekit-devices#2336] [MEPLHAA Scripts]
Hi everyone, I write this script to setup a son off 4 pro with 3 water valve and one switch, pressing the buttons 1,2,3 I can move the relative out but the 4th button doesn't move nothing. By the app I can move out4

{"c":
{"io":[[[4,5,12,15,13],2],[[0,9,10,14],6]],"l":13,"b":[[0,5]]},
"a":[
{"t":20,"d":0,"0":{"r":[[12]]},"1":{"r":[[12,1]]},"b":[[0]]},
{"t":20,"0":{"r":[[5]]},"1":{"r":[[5,1]]},"b":[[9]]},
{"t":20,"0":{"r":[[4]]},"1":{"r":[[4,1]]},"b":[[10]]},
{"0":{"r":[[15]]},"1":{"r":[[15,1]]},"b":[[14]]}
]}

**Comment by RavenSystem** [comment]
Try this, enabling internal pullup resistor for GPIO 14:
```json
{"c":
{"io":[[[4,5,12,15,13],2],[[0,9,10],6],[[14],6,1]],"l":13,"b":[[0,5]]},
"a":[
{"t":20,"d":0,"0":{"r":[[12]]},"1":{"r":[[12,1]]},"b":[[0]]},
{"t":20,"0":{"r":[[5]]},"1":{"r":[[5,1]]},"b":[[9]]},
{"t":20,"0":{"r":[[4]]},"1":{"r":[[4,1]]},"b":[[10]]},
{"0":{"r":[[15]]},"1":{"r":[[15,1]]},"b":[[14]]}
]}
```

**Reply by crazyclimber999** [reply]
Thank you so much, it works.

**Reply by RavenSystem** [reply]
Thanks! I fixed it in the Wiki database.
